jobs/env/ci-kubernetes-kubemark-gce-big.env

```diff
@@ -6,4 +6,4 @@ KUBEMARK_MASTER_COMPONENTS_QPS_LIMITS=--kube-api-qps=100 --kube-api-burst=100
 
 ### e2e-env
 # Increase throughput in Load test.
-LOAD_TEST_THROUGHPUT=50
+LOAD_TEST_THROUGHPUT=100
```

Comment:
You would need to increase qps-limits, otherwise we will stuck, I think....

Reply:
What I mean is that single update e.g. requires get + update calls.

Reply:
Shouldn't such gets be served from informer stores? Are there any particular exceptions you're referring to?

Reply:
I can't remember from the top of my head - but I'm pretty sure there are a number of places where throughput needs to be lower than qps. You can take a look into the code.

Reply:
Also, this should still not cause failures iiuc (only thing is server-side processing could still be bottlenecked). No?

Reply:
Btw, from the perspective of our test we're only doing 2 things:

- creating pods (during creation and scaling) - And this one I'm quite sure can already happen at 100/s
- deleting pods (during scaling and deletion) - This one like we discussed earlier can only happen at 50/s

In any case, this PR should help with creation speed and be ~no-op with deletion speed. Wdyt?

Reply:
Significant bottleneck may may timeouts on starting pods (e.g. in load tests) - it will be taking much longer. I'm almost 100% sure that it actually will cause problems.

Reply:
... if you won't bumpt qps limits

Reply:
> Significant bottleneck may may timeouts on starting pods (e.g. in load tests) - it will be taking much longer.

Could you clarify why it'll take longer? We're already creating pods at that rate in density test right?

Reply:
The test itself will not take longer. It will take longer to start an individual RC (because there will be backlog to process first for those created at the end).

Reply:
True for creations - not true for updates and deletions IIRC.

Reply:
For deletions, what you say is valid (but that's easily fixable).

Reply:
(sorry.. race)
Yeah, I can fix that for deletions. And I'm not sure if we're doing any pod updates in our tests.

Reply:
PTAL https://github.com/kubernetes/kubernetes/pull/67776

Reply:
I'm not saying about pod updated - I'm saying about RC updates (or more specifically scaling them).
on average ~half of scalings is resulting in pod deletions - so the argument above.

That is also a comment to kubernetes/kubernetes#67776 as a result.

Reply:
I was expecting you to ask this question :)

So IIUC given that the deletes happening in the scaling step are performed by RC-controller (as opposed to GC), we should be able to achieve full throughput as we don't do the extra GET call there.

Reply:
> I'm not saying about pod updated - I'm saying about RC updates (or more specifically scaling them).

I wouldn't really be calling that an update in this context (where we're talking about pods) as RC update is O(1) operation.

Reply:
So to clarify - I wasn't worried about RCs updates - that I agree is O(1).

Re deletions of pods - yeah - it seems that is a single call (+ event, but that's a separate qps limit).

So maybe that's actually fine. Let's try.

But I think we need to merge the k/k PR first.

Reply:
> But I think we need to merge the k/k PR first.

Maybe lgtm and hold? (I'll cancel it later)